backend(oidc): Add support for custom CA and skipping TLS verification

yolossn opened the pull request:
this patch adds support for custom CA and skipping TLS verification for OIDC provider verification. This improves flexibility for users working with self-signed OIDC providers.

Fixes: 
https://github.com/kubernetes-sigs/headlamp/issues/127 
https://github.com/kubernetes-sigs/headlamp/issues/3707
https://github.com/kubernetes-sigs/headlamp/issues/2207 (part 1)
> OIDC doesn't support self signed CA - Going through the instructions for OIDC ignores trusting a certificate. This makes OIDC mostly useless for identity prociders that aren't publicly hosted, like OpenUnison, in an enterprise environment which usually has its own internal CA.


Steps to test:
1. Deploy Dex or any other OIDC provider with custom CA
2. Configure the minikube cluster to use OIDC auth with your OIDC provider with custom CA
3. Deploy headlamp without skip-tls-verify or customCAs, you will the following error when you click "Sign In"
<img width="1512" height="982" alt="image" src="https://github.com/user-attachments/assets/fb91a2cd-e275-455e-aa80-6bce46dd6eea" />
4. Deploy headlamp with skip-tls-verify alone configure. You will get a warning message that this isn't a safe way to use and suggest not do this in production.
<img width="1358" height="163" alt="image" src="https://github.com/user-attachments/assets/823d98e8-d5cc-4e95-800b-6bdba9f3cfb2" />
On clicking Sign In you should be able to see the OIDC providers login page
<img width="1512" height="982" alt="image" src="https://github.com/user-attachments/assets/26eb6bc0-d4be-415d-afe3-a870d78900d6" />
5. Deploy headlamp with customCA configured, you will not see the warning and the sign in should work.
<img width="1358" height="163" alt="image" src="https://github.com/user-attachments/assets/cfd6dd52-f16b-4a56-859e-e08e0150643e" />
<img width="1512" height="982" alt="image" src="https://github.com/user-attachments/assets/3ca3f930-8444-4cf6-ac72-6a016052fbc2" />







Note: 
re-opened based on https://github.com/kubernetes-sigs/headlamp/pull/2880 due to branch restrictions.

yolossn committed:
backend(oidc): Add support for custom CA and skipping TLS verification

this patch adds support for custom CA and skipping TLS verification
for OIDC provider verification. This improves flexibility for users
working with self-signed OIDC providers.

Signed-off-by: yolossn <sannagaraj@microsoft.com>

illume reviewed (changes requested):
This note from the last PR:

https://github.com/kubernetes-sigs/headlamp/pull/2880#pullrequestreview-2607593252
> I left some notes inline.
> 
> * Can you please add tests and a description of how to test manually in the PR?
> * Probably the related issue(s) should be in the PR description.
> * There will need to be support added to the helm chart.
> * docs in oidc.md ?

I didn't check if the notes after this made in the other PR were addressed here. Were they?

yolossn committed:
docs: Add self-signed CA usage documentation for OIDC

Adds documentation for using self-signed CA certificates with OIDC in Headlamp,
including command line options, kubeconfig configuration, and examples.

Signed-off-by: yolossn <sannagaraj@microsoft.com>

illume reviewed (approved):
Great work. 🎉 thanks!

(Ultimately we need to add support for at least one of these flags to the helm chart. But I think we can merge this in now, and that be done in a separate PR?)

yolossn commented:
@illume  Thanks for the review, I will create a follow up PR for the helm chart and update the `oidc.md` to include the helm charts too. Would like to get this merged for now.
